lua/wire/flir.lua

```diff
@@ -20,41 +21,100 @@ if CLIENT then
 	FLIR.normal = CreateMaterial("flir_normal", "VertexLitGeneric", {
 		["$basetexture"] = "color/white",
 		["$model"] = 1,
-		["$halflambert"] = 1 -- causes the diffuse lighting to 'wrap around' more
+		["$halflambert"] = 1, -- causes the diffuse lighting to 'wrap around' more
+		["$color2"] = "[10.0 10.0 10.0]"
 	})
 
-	FLIR.colmod = {
-		[ "$pp_colour_addr" ] = 0.4,
-		[ "$pp_colour_addg" ] = -.5,
-		[ "$pp_colour_addb" ] = -.5,
-		[ "$pp_colour_brightness" ] = .1,
-		[ "$pp_colour_contrast" ] = 1.2,
-		[ "$pp_colour_colour" ] = 0,
-		[ "$pp_colour_mulr" ] = 0,
-		[ "$pp_colour_mulg" ] = 0,
-		[ "$pp_colour_mulb" ] = 0
+	FLIR.entcol = {
+		["$pp_colour_colour" ] = 0,
+		["$pp_colour_brightness"] = -0.3,
+		["$pp_colour_contrast"] = 1.0
+	}
+
+	FLIR.mapcol = {
+		[ "$pp_colour_brightness" ] = 0.8,
+		[ "$pp_colour_contrast" ] = 0.5,
+		[ "$pp_colour_colour"] = 0.0,
+	}
+
+	FLIR.skycol = {
+		[ "$pp_colour_brightness" ] = -0.3
 	}
 
 	local materialOverrides = {
 			PlayerDraw = { FLIR.living, FLIR.normal },
-			DrawOpaqueRenderables = { FLIR.normal, nil },
-			DrawTranslucentRenderables = { FLIR.normal, nil },
 			DrawSkybox = { FLIR.normal, nil }
 	}
 
+	local function SetFLIRMat(ent)
+		if not IsValid(ent) then return end
+
+		if ent:GetMoveType() == MOVETYPE_VPHYSICS or IsValid(ent:GetParent()) or ent:IsPlayer() or ent:IsNPC() or ent:IsRagdoll() then
+				ent.FLIRMat = ent:GetMaterial()
+				ent:SetMaterial("!flir_normal")	
```

Comment:
I don't think setting the entities material is a good solution. Any changes in material while flir is enabled will not be accounted for and then when flir is disabled, the material changes will be reverted to before flir was enabled.

Reply:
You're right that it isn't a good solution at all. Problem is, i tried every better solution i could find or think of and couldn't find anything that worked to separate props from other opaque renderables (which drastically improves the look).

I don't know. Maybe I can work a solution to allow materials to be changed correctly, but gmod is pretty janky about all of that unfortunately.

Reply:
I can't think of any better way but I'm not great at the rendering pipeline. This can be improved later if needed. Mind implementing the other review comment?

Reply:
@kerbalcadet

Reply:
@thegrb93 

Sorry for my absence. College will do that to you. Anyhow, I updated and slightly reworked this whole thing to work out all the problems I could.